dash/development/_r_components_generation.py

```diff
@@ -259,7 +262,8 @@ def generate_js_metadata(pkg_data, project_shortname):
     sys.path.insert(0, os.getcwd())
     mod = importlib.import_module(project_shortname)
 
-    alldist = getattr(mod, "_js_dist", []) + getattr(mod, "_css_dist", [])
+    alldist = getattr(mod, "_js_dist", []) + getattr(mod, "_css_dist", []) \
+        + getattr(mod, "_asset_dist", [])
```

Comment:
We've discussed an `_asset_dist` bucket that would generalize on the existing `_js_dist` and `_css_dist` ones. Is this work towards that direction or are the arbitrary file extensions required to be in `_asset_dist`? The Python PR does not have an equivalent for this atm.

Reply:
I think @wbrgss suggested we discuss this with you, but I'm fine with `_asset_dist` as a general bucket if it works elsewhere. If we should use another structure, just let me know -- either works for me.

Reply:
> @rpkyle I think we should drop `_asset_dist` and just use `_js_dist` for now. While the `_asset_dist` is an interesting addition, I think we should rework css/js/asset in depth and with all Dash flavors in mind before committing to supporting additional props. `_js_dist` is not perfect but it covers all usages with `_css_dist`
> 
> ~This PR~ Portions of this PR would be superseded by #1078.

OK, I'll remove the elements of this pull request which are addressed by #1078. The remaining two items are still worth merging in for the sake of the CRAN release.